server/src/main/java/org/apache/seata/server/cluster/raft/manager/RaftControllerServerManager.java

```diff
@@ -0,0 +1,181 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.seata.server.cluster.raft.manager;
+
+
+import com.alipay.sofa.jraft.entity.PeerId;
+import com.alipay.sofa.jraft.option.NodeOptions;
+import com.alipay.sofa.jraft.rpc.RpcServer;
+import org.apache.seata.server.cluster.raft.AbstractRaftServerManager;
+import org.apache.seata.server.cluster.raft.RaftServerConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicReference;
+
+/**
+ * Controller Raft Server Manager for managing metadata across multiple raft groups
+ */
+public class RaftControllerServerManager extends AbstractRaftServerManager<RaftControllerServer> {
+
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(RaftControllerServerManager.class);
+
+    // Map of controller group IDs to RaftControllerServer instances
+    private static final Map<String /*controller-group*/, RaftControllerServer /*raft-controller-cluster*/> RAFT_CONTROLLER_SERVER_MAP = new HashMap<>();
+    private static volatile boolean CONTROLLER_MODE;
+
+    // Default controller group name
+    private static final String DEFAULT_CONTROLLER_GROUP = "controller-group";
+
+    @Override
+    protected RaftServerConfig getRaftServerConfig() {
+        return new RaftServerConfig(
+                "server.raft.controller.cluster",
+                "server.raftControllerPort",
+                "server.raft.controller.group",
+                DEFAULT_CONTROLLER_GROUP,
+                "raft-controller",
+                "server.raft.controller.cluster has duplicate ip, For local debugging, use -Dserver.raftControllerPort to specify the raft controller port"
+        );
+    }
+
+    @Override
+    public void init() {
+        super.init();
+    }
+
+    @Override
+    protected RaftControllerServer createRaftServer(String dataPath, String group, PeerId serverId,
+                                                    NodeOptions nodeOptions, RpcServer rpcServer) throws IOException {
+        return new RaftControllerServer(dataPath, group, serverId, nodeOptions, rpcServer);
+    }
+
+    @Override
+    protected void storeRaftServer(String group, RaftControllerServer raftControllerServer) {
+        RAFT_CONTROLLER_SERVER_MAP.put(group, raftControllerServer);
+    }
+
+    @Override
+    protected boolean isModeEnabled() {
+        CONTROLLER_MODE = CONFIG.getBoolean("server.raft.controller.enabled", false);
```

Comment:
don't understand why this configuration is needed. If the user doesn't configure the controller's configuration item, doesn't that mean it's disabled?